console/console-init/ui/src/modules/device/components/CredentialList/CredentialList.tsx

```diff
@@ -166,30 +182,35 @@ export const CredentialList: React.FC<ICredentialListProps> = ({
             isExpandedAdvancedSetting = false,
             type = ""
           } = credential;
-          const isSecretsHeadingVisible = type === "x509" ? false : true;
           return (
             <Grid key={id}>
               <GridItem span={6}>
                 <Card>
                   <CardHead>
                     <CardActions>
-                      <Button
-                        variant="link"
-                        type="button"
-                        onClick={() => onDeleteItem(id)}
-                        icon={<ErrorCircleOIcon />}
-                      />
+                      {credentials.length > 1 && (
```

Comment:
I think it should be credentials.length>0 ?

Reply:
That is correct. let me explain, delete credential icon will display in case we have more than one credentials, in not then we will not show delete option as per design.